### DISCUSSION TITLE
Allow to ignore a sub-recipe

### DISCUSSION THREAD

**Discussion by adriil** [openrewrite/rewrite#3252] [Q&A] [answered]
Hi team,

Let's say I have a complex Spring Boot project that I would like to migrate to Spring Boot 2.7 or Spring Boot 3, **but** I don't want to migrate from `JUnit 4` to `JUnit 5` like it is done by the `Migrate to Spring Boot 2.4` recipe.

Is there a way to activate a recipe **and** define sub-recipe(s) that we'd like to ignore, like the following ?
```
rewrite {
    activeRecipe("org.openrewrite.java.spring.boot2.UpgradeSpringBoot_2_7") {
        ignore("org.openrewrite.java.spring.boot2.SpringBoot2JUnit4to5Migration")
    }
}
```

Thanks,
Adrien

**Comment by timtebeek** [comment, marked as answer]
Hi @adriil ; There's not currently a way to exclude certain recipes, as it get's complicated quickly with the way recipes chain one another, and the assumptions they each build upon in having the code in a certain state.

That said, for your particular use case here it sounds like you'd be interested in the work started in https://github.com/openrewrite/rewrite-spring/pull/351 ; there we're considering having separate recipes that decouple the spring migrations from one another, such that you can run the individual spring minor version migrations without getting the chained junit 5 migration with each. Would that help you through here?

As a side note: I think Spring Boot 3 dropped support for JUnit 4, at least for the spring specific test constructs. Up until Spring Boot 2.7 you can still get away with the junit vintage runner I believe, so you might get stuck at 2.7 with vintage runner for JUnit 4. Ultimately we do recommend you upgrade fully to JUnit 5; let us know if you're missing specific functionality to migrate fully.
